api/v1alpha1/maasvalidator_types.go

```diff
@@ -22,47 +22,96 @@ import (
 
 // MaasValidatorSpec defines the desired state of MaasValidator
 type MaasValidatorSpec struct {
-	MaasInstanceRules `json:"MaasInstanceRules,omitempty" yaml:"MaasInstanceRules,omitempty"`
-	MaasInstance      `json:"MaasInstance" yaml:"MaasInstance"`
+	Auth Auth   `json:"auth" yaml:"auth"`
+	Host string `json:"host" yaml:"host"`
+
+	OSImageRules              []OSImageRule              `json:"osImageRules,omitempty" yaml:"osImageRules,omitempty"`
```

Comment:
Sorry, but let's just call these `ImageRules`, since the MAAS API just refers to them as "Images", not OSImages.